bin/propolis-server/src/lib/stats.rs

```diff
@@ -46,24 +51,12 @@ const VCPU_KSTAT_INTERVAL: std::time::Duration =
 //
 // This limit provides extra space for up to 64 samples per vCPU per microstate,
 // to ensure we don't throw away too much data if oximeter cannot reach us.
-#[cfg(all(not(test), target_os = "illumos"))]
 const KSTAT_LIMIT_PER_VCPU: u32 =
     crate::stats::virtual_machine::N_VCPU_MICROSTATES * 64;
 
-/// An Oximeter `Metric` that specifies the number of times an instance was
-/// reset via the server API.
-#[derive(Debug, Default, Copy, Clone, Metric)]
-struct Reset {
-    /// The number of times this instance was reset via the API.
-    #[datum]
-    pub count: Cumulative<u64>,
-}
-
-/// The full set of server-level metrics, collated by
-/// [`ServerStatsOuter::produce`] into the types needed to relay these
-/// statistics to Oximeter.
+/// Shared type for tracking metrics about the Propolis API server itself.
 #[derive(Clone, Debug)]
-struct ServerStats {
+struct ServerStatsInner {
```

Comment:
I was expecting it to go away as well. But the `Reset` metric _is_ currently published to `oximeter`, so I think some form of this needs to stick around. At that point, it's similar to the current block stats: there's some state shared between the Propolis server, which updates the reset count, and the `oximeter_producer::Server`, which publishes the value when `oximeter` polls for it.

It's possible it could be shared differently, so I'll take another look at it.

Reply:
Thanks! Don't spend too much time on it; I was mostly just curious about how this had shaken out in practice.

Reply:
I think this is in a reasonable place as-is. Cleaning up the naming and the initialization process makes it much clearer that the `ServerStats` type is used to store metrics about the API server or requests it services. We could add HTTP latency tracking here for example, if we cared to know that. Thanks for flagging it!